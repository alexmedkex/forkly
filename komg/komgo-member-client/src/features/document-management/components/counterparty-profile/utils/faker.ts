import { CounterpartyProfile, RiskLevel } from '../../../../counterparties/store/types'

const defaultProfile: CounterpartyProfile = {
  id: 'anon-profile-id',
  counterpartyId: 'anon-counterparty-id',
  renewalDate: new Date('2019-07-16').toISOString(),
  riskLevel: RiskLevel.unspecified,
  managedById: 'anon-user-id'
}

export const fakeProfile = (partial?: Partial<CounterpartyProfile>) => {
  return partial ? { ...defaultProfile, ...partial } : defaultProfile
}

export const mockDate = () => {
  const RealDate = Date
  const _GLOBAL: any = global // we love typescript!

  return {
    freeze: (isoDate: string | number | Date) => {
      _GLOBAL.Date = class extends RealDate {
        constructor(...args: any) {
          super()
          return new RealDate(isoDate)
        }
      }
    },
    restore: () => {
      global.Date = RealDate
    }
  }
}
