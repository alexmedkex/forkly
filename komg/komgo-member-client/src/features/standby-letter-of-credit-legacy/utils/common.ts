import { IStandbyLetterOfCredit, StandbyLetterOfCreditStatus } from '@komgo/types'
export const isRejectedStandbyLetterOfCredit = (letter: IStandbyLetterOfCredit) => {
  return (
    !!letter &&
    [StandbyLetterOfCreditStatus.RequestRejected, StandbyLetterOfCreditStatus.IssuedRejected].includes(letter.status)
  )
}
