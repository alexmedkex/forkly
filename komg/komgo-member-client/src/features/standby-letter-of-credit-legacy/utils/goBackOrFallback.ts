import { history } from '../../../store'

export const goBackOrFallBackTo = ({
  defaultHistoryLength = 4, // In KOMGO initial history.length is by default 4 because the login redirects and route configuration
  fallbackURL
}: {
  defaultHistoryLength?: number
  fallbackURL: string
}) => {
  // WORKAROUND check if the history length is the default defaultHistoryLength
  // TODO LS remove anything that mimic browser behaviors (e.g. back). This approach is fragile
  if (history.length <= defaultHistoryLength) {
    history.push(fallbackURL)
  } else {
    history.goBack()
  }
}
