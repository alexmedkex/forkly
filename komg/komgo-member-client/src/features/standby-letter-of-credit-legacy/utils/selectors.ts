import { Task, TaskStatus } from '../../tasks/store/types'

export const findTasksByStandByLetterOfCreditAndStatuses = (
  letterId: string,
  tasks: Task[],
  statuses: TaskStatus[]
) => {
  return tasks.filter(task => task.context && task.context.sblcStaticId === letterId && statuses.includes(task.status))
}
