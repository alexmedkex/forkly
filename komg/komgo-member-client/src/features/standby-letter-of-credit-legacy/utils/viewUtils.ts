import { ReviewDecision, IssueFormValues } from '../components/issue-form'
import { StandbyLetterOfCreditTaskType, IStandbyLetterOfCredit } from '@komgo/types'
import { IMember } from '../../members/store/types'

export const findActiveFields = (
  taskType: StandbyLetterOfCreditTaskType,
  reviewDecision: ReviewDecision
): Array<Partial<keyof IStandbyLetterOfCredit>> =>
  taskType === StandbyLetterOfCreditTaskType.ReviewRequested && reviewDecision === ReviewDecision.ApproveApplication
    ? ['issuingBankReference', 'issuingBankPostalAddress']
    : []

const isSet = (val?: string) => !!val

// cyclomatic complexity....
const isApproveApplicationUseCase = (
  taskIsReviewRequested: boolean,
  reviewDecision: ReviewDecision,
  hasIssuingBankReference: boolean,
  hasIssuingBankPostalAddress: boolean,
  beneficiary: IMember,
  document?: File
) => {
  const rules = [
    taskIsReviewRequested &&
      reviewDecision === ReviewDecision.ApproveApplication &&
      hasIssuingBankReference &&
      hasIssuingBankPostalAddress &&
      beneficiary.isMember,
    taskIsReviewRequested &&
      reviewDecision === ReviewDecision.ApproveApplication &&
      !beneficiary.isMember &&
      document &&
      hasIssuingBankReference &&
      hasIssuingBankPostalAddress
  ]

  return rules.includes(true)
}

export const isSubmitResponseEnabled = (
  values: IssueFormValues,
  beneficiary: IMember,
  taskType: StandbyLetterOfCreditTaskType
) => {
  const { document, reviewDecision, standbyLetterOfCredit } = values

  const taskIsReviewRequested = taskType === StandbyLetterOfCreditTaskType.ReviewRequested

  const hasIssuingBankPostalAddress = isSet(standbyLetterOfCredit.issuingBankPostalAddress)
  const hasIssuingBankReference = isSet(standbyLetterOfCredit.issuingBankReference)

  const rules = [
    taskIsReviewRequested && reviewDecision === ReviewDecision.RejectApplication,
    isApproveApplicationUseCase(
      taskIsReviewRequested,
      reviewDecision,
      hasIssuingBankReference,
      hasIssuingBankPostalAddress,
      beneficiary,
      document
    )
  ]
  return rules.includes(true)
}

export const viewSubmitHandler = (
  values: IssueFormValues,
  taskType: StandbyLetterOfCreditTaskType,
  issueStandbyLetterOfCredit: (standbyLetterOfCredit: IStandbyLetterOfCredit, file: File) => void,
  rejectStandbyLetterOfCreditRequest: (staticId: string, issuingBankReference: string) => void
) => {
  const taskIsReviewRequested = taskType === StandbyLetterOfCreditTaskType.ReviewRequested

  if (taskIsReviewRequested && values.reviewDecision === ReviewDecision.ApproveApplication) {
    issueStandbyLetterOfCredit(values.standbyLetterOfCredit, values.document)
  } else if (taskIsReviewRequested && values.reviewDecision === ReviewDecision.RejectApplication) {
    rejectStandbyLetterOfCreditRequest(values.standbyLetterOfCredit.staticId, values.rejectionReference)
  }
}
