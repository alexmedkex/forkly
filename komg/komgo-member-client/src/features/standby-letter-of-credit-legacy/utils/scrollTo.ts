import scrollIntoView from 'scroll-into-view'
import { history } from '../../../store/history'
import { SIZES } from '../../../utils/media'

export const scrollTo = (selector: string) => {
  const isMobile = window.innerWidth < SIZES.desktop

  if (isMobile) {
    return
  }

  const target = document.querySelector(selector)
  scrollIntoView(target, { time: 500, align: { top: 0.1 } })
}
