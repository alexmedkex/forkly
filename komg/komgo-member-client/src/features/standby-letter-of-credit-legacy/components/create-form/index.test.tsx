import * as React from 'react'
import { mount, shallow } from 'enzyme'
import { Field } from 'formik'
import { MemoryRouter as Router } from 'react-router-dom'
import * as renderer from 'react-test-renderer'
import { CreateForm, CreateFormProps } from './index'
import { fakeCounterparty, fakeMember } from '../../../letter-of-credit-legacy/utils/faker'
import {
  IStandbyLetterOfCreditBase,
  TradeSource,
  DuplicateClause,
  CompanyRoles,
  Currency,
  Fee,
  buildFakeStandByLetterOfCreditBase
} from '@komgo/types'
import { Dropdown } from 'semantic-ui-react'
import { scrollTo } from '../../utils/scrollTo'
jest.mock('../../utils/scrollTo')

describe('CreateForm', () => {
  let props: CreateFormProps

  beforeEach(() => {
    props = {
      initialValues: buildFakeStandByLetterOfCreditBase({
        tradeId: { source: TradeSource.Komgo, sourceId: '55918db4-8a94-495c-a9b3-0aafebc74328' },
        issuingBankId: undefined,
        beneficiaryBankId: undefined,
        beneficiaryBankRole: undefined,
        amount: 1000,
        currency: Currency.USD,
        contractReference: undefined,
        contractDate: undefined,
        duplicateClause: DuplicateClause.Yes,
        expiryDate: undefined,
        additionalInformation: undefined,
        overrideStandardTemplate: 'Default legal template clauses',
        availableWith: CompanyRoles.IssuingBank,
        feesPayableBy: Fee.Split
      }),
      onSubmit: jest.fn(),
      onChange: jest.fn(),
      issuingBanks: [
        fakeCounterparty({ staticId: 'abc1111', commonName: 'bank1', isMember: true, isFinancialInstitution: true }),
        fakeCounterparty({ staticId: 'abc2222', commonName: 'bank2', isMember: true, isFinancialInstitution: true }),
        fakeCounterparty({ staticId: 'abc3333', commonName: 'bank3', isMember: true, isFinancialInstitution: true })
      ],
      beneficiaryBanks: [
        fakeMember({ staticId: 'abc1111', commonName: 'bank1', isMember: true, isFinancialInstitution: true }),
        fakeMember({ staticId: 'abc2222', commonName: 'bank2', isMember: true, isFinancialInstitution: true }),
        fakeMember({ staticId: 'abc3333', commonName: 'bank3', isMember: true, isFinancialInstitution: true })
      ]
    }
  })

  describe('render', () => {
    it('shows the default values', () => {
      Date.now = jest.fn(() => 1487076708000)

      expect(
        renderer
          .create(
            <Router>
              <CreateForm {...props} />
            </Router>
          )
          .toJSON()
      ).toMatchSnapshot()
    })
  })

  describe('scrollTo', () => {
    it('scrolls to contractReference label', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find('input[id="field_contractReference"]').simulate('focus')
      expect(scrollTo).toHaveBeenCalledWith('#preview_contractReference')
    })

    it('scrolls to contractDate label', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find('input[id="field_contractDate"]').simulate('focus')
      expect(scrollTo).toHaveBeenCalledWith('#preview_contractDate')
    })

    it('scrolls to issuingBankId label', done => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find('div[name="issuingBankId"]').simulate('focus')
      setTimeout(() => {
        expect(scrollTo).toHaveBeenCalledWith('#preview_issuingBankId')
        done()
      }, 0)
    })

    it('scrolls to amount label', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find('input[id="field_amount"]').simulate('focus')
      expect(scrollTo).toHaveBeenCalledWith('#preview_amount')
    })

    it('scrolls to expiryDate label', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find('input[id="field_expiryDate"]').simulate('focus')
      expect(scrollTo).toHaveBeenCalledWith('#preview_expiryDate')
    })

    it('scrolls to feesPayableBy label', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find(`div[data-test-id="feesPayableBy_${Fee.Applicant}"]`).simulate('focus')
      expect(scrollTo).toHaveBeenCalledWith('#preview_feesPayableBy')
    })

    it('scrolls to duplicateClause label', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find(`div[data-test-id="duplicateClause_${DuplicateClause.Yes}"]`).simulate('focus')
      expect(scrollTo).toHaveBeenCalledWith('#preview_duplicateClause')
    })

    it('scrolls to overrideStandardTemplate label', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find('textarea[id="field_overrideStandardTemplate"]').simulate('focus')
      expect(scrollTo).toHaveBeenCalledWith('#preview_overrideStandardTemplate')
    })
  })

  describe('onSubmit', () => {
    it('sends the value collected', done => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper.find('form').simulate('submit')

      setTimeout(() => {
        expect(props.onSubmit).toHaveBeenCalledWith(props.initialValues)
        done()
      })
    })
  })

  describe('onChange', () => {
    it('types a contractReference', () => {
      const wrapper = mount(<CreateForm {...props} />)
      const value = 'REF-123'
      wrapper
        .find('input[name="contractReference"]')
        .simulate('change', { target: { value, name: 'contractReference' } })

      expect(wrapper.find('input[name="contractReference"]').prop('value')).toEqual(value)

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        contractReference: value
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })

    it('types a contractDate', () => {
      const wrapper = mount(<CreateForm {...props} />)
      const value = '2019-04-01'
      wrapper.find('input[name="contractReference"]').simulate('change', { target: { value, name: 'contractDate' } })

      expect(wrapper.find('input[name="contractDate"]').prop('value')).toEqual(value)

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        contractDate: value
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })

    it('selects an issuingBank', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper
        .find(Field)
        .find({ name: 'issuingBankId' })
        .find(Dropdown.Item)
        .first()
        .simulate('click')

      expect(
        wrapper
          .find({ name: 'issuingBankId' })
          .find(Dropdown)
          .prop('value')
      ).toEqual('abc1111')

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        issuingBankId: 'abc1111'
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })

    // TODO LS removed from the scope
    it.skip('selects a beneficiaryBank', () => {
      const wrapper = mount(<CreateForm {...props} />)
      wrapper
        .find(Field)
        .find({ name: 'beneficiaryBankId' })
        .find(Dropdown.Item)
        .at(2)
        .simulate('click')

      expect(
        wrapper
          .find({ name: 'beneficiaryBankId' })
          .find(Dropdown)
          .prop('value')
      ).toEqual('abc2222')

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        beneficiaryBankId: 'abc2222'
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })
    it('set an expiryDate', () => {
      const wrapper = mount(<CreateForm {...props} />)
      const today = '2019-04-01'
      wrapper.find('input[name="expiryDate"]').simulate('change', { target: { value: today, name: 'expiryDate' } })

      expect(wrapper.find('input[name="expiryDate"]').prop('value')).toEqual(today)

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        expiryDate: today
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })

    it('selects a fee', () => {
      const wrapper = mount(<CreateForm {...props} />)
      const value = Fee.Applicant

      wrapper.find(`div[data-test-id="feesPayableBy_${value}"]`).simulate('click')

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        feesPayableBy: value
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })

    it('selects duplicate clause', () => {
      const wrapper = mount(<CreateForm {...props} />)
      const value = DuplicateClause.No

      wrapper.find(`div[data-test-id="duplicateClause_${value}"]`).simulate('click')

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        duplicateClause: value
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })

    it('types a custom template', () => {
      const wrapper = mount(<CreateForm {...props} />)
      const value = 'custom template'
      wrapper
        .find('textarea[name="overrideStandardTemplate"]')
        .simulate('change', { target: { value, name: 'overrideStandardTemplate' } })

      expect(wrapper.find('textarea[name="overrideStandardTemplate"]').prop('value')).toEqual(value)

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        overrideStandardTemplate: value
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })

    it('types an additional information', () => {
      const wrapper = mount(<CreateForm {...props} />)
      const value = 'additional information'
      wrapper
        .find('textarea[name="additionalInformation"]')
        .simulate('change', { target: { value, name: 'additionalInformation' } })

      expect(wrapper.find('textarea[name="additionalInformation"]').prop('value')).toEqual(value)

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        additionalInformation: value
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })
  })

  describe('onBlur', () => {
    it('updates amount', () => {
      const wrapper = mount(<CreateForm {...props} />)

      wrapper
        .find('input[name="amount"]')
        .simulate('focus')
        .simulate('change', {
          target: { value: '100000' }
        })
        .simulate('blur')

      expect(wrapper.find('input[name="amount"]').prop('value')).toEqual('100,000.00')

      const letter: IStandbyLetterOfCreditBase = {
        ...props.initialValues,
        amount: 100000
      }

      expect(props.onChange).toHaveBeenCalledWith(letter)
    })
    describe('validation error', () => {
      it('is shown if the contractReference is left empty', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('input[name="contractReference"]')
          .simulate('change', { target: { value: '', name: 'contractReference' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="contractReference-error"]').text()).not.toEqual('')
          done()
        })
      })
      // TODO RR NOT SURE WHY IT DOESN'T WORK
      it.skip('shows the card at the top if a mandatory field is left empty', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('input[name="contractReference"]')
          .simulate('change', { target: { value: '', name: 'contractReference' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="requiredFieldPrompt"]').text()).toEqual(
            'Please complete all required fields'
          )
          done()
        })
      })
      it('is shown if the contractDate is left empty', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('input[name="contractDate"]')
          .simulate('change', { target: { value: '', name: 'contractDate' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="contractDate-error"]').text()).not.toEqual('')
          done()
        })
      })
      it('is shown if the issuingBankId is left empty', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('div[name="issuingBankId"]')
          .simulate('change', { target: { value: '', name: 'issuingBankId' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="issuingBankId-error"]').text()).not.toEqual('')
          done()
        })
      })
      it('is shown if the amount is left empty', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('input[name="amount"]')
          .simulate('change', { target: { value: '', name: 'amount' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="amount-error"]').text()).not.toEqual('')
          done()
        })
      })
      it('is shown if the expiryDate is left empty', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('input[name="expiryDate"]')
          .simulate('change', { target: { value: '', name: 'expiryDate' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="expiryDate-error"]').text()).not.toEqual('')
          done()
        })
      })
      it('is shown if the expiryDate is is the past', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('input[name="expiryDate"]')
          .simulate('change', { target: { value: '2017-04-10', name: 'expiryDate' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="expiryDate-error"]').text()).not.toEqual('')
          done()
        })
      })
      it('is shown if the overrideStandardTemplate is left empty', done => {
        const wrapper = mount(<CreateForm {...props} />)
        wrapper
          .find('textarea[name="overrideStandardTemplate"]')
          .simulate('change', { target: { value: '', name: 'overrideStandardTemplate' } })
          .simulate('blur')

        setTimeout(() => {
          expect(wrapper.find('div[data-test-id="overrideStandardTemplate-error"]').text()).not.toEqual('')
          done()
        })
      })
    })
  })
})
