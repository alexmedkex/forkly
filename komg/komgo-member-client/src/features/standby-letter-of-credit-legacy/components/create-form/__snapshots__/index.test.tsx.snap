// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateForm render shows the default values 1`] = `
.c0 {
  font-size: small;
  color: #d45c64;
  margin-top: -1em;
  padding-top: 3px;
  min-height: 21px;
  margin-bottom: 9px;
}

<form
  className="ui form"
  id="create-standby-letter-of-credit"
  onSubmit={[Function]}
>
  <div
    className="ui fluid card"
    data-test-id="requiredFieldPrompt"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#d45c64",
        "border": "none",
        "color": "white",
        "textAlign": "center",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="content"
    >
      Please complete all required fields
    </div>
  </div>
  <div
    className="field"
    style={
      Object {
        "alignItems": "center",
        "display": "inherit",
        "justifyContent": "unset",
      }
    }
  >
    <label
      className="inputLabel"
    >
      Contract reference
    </label>
    <div
      className="ui input"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <input
        id="field_contractReference"
        name="contractReference"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex="1"
        type="text"
        value="CREF-123"
      />
    </div>
  </div>
  <div
    className="c0"
    data-test-id="contractReference-error"
  >
    
  </div>
  <div
    className="field"
    style={
      Object {
        "alignItems": "center",
        "display": "inherit",
        "justifyContent": "unset",
      }
    }
  >
    <label
      className="inputLabel"
    >
      Contract date
    </label>
    <div
      className="ui input"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <input
        id="field_contractDate"
        name="contractDate"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex="2"
        type="date"
        value="2019-10-20"
      />
    </div>
  </div>
  <div
    className="c0"
    data-test-id="contractDate-error"
  >
    
  </div>
  <div
    className="field"
  >
    <label
      className="inputLabel"
    >
      Issuing bank
    </label>
    <div
      aria-expanded={false}
      className="ui compact search selection dropdown dropdown-input"
      name="issuingBankId"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex="3"
        type="text"
        value=""
      />
      <div
        aria-live="polite"
        className="text"
        role="alert"
      />
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
        role="listbox"
      >
        <div
          aria-checked={false}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            bank1
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            bank2
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            bank3
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c0"
    data-test-id="issuingBankId-error"
  >
    
  </div>
  <div
    className="field"
  >
    <label
      className="inputLabel"
    >
      Beneficiary bank
    </label>
    <div
      className="ui disabled input field disabled"
      search={true}
      selection={true}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <input
        disabled={true}
        name="beneficiaryBankId"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="No Advising Bank"
        tabIndex="4"
        type="text"
        value="a28b8dc3-8de9-4559-8ca1-272ccef52b47"
      />
    </div>
  </div>
  <div
    className="c0"
    data-test-id="beneficiaryBankId-error"
  >
    
  </div>
  <div
    className="field"
  >
    <label
      className="inputLabel"
    >
      Available with
    </label>
    <div
      aria-expanded={false}
      className="ui compact search selection dropdown dropdown-input"
      name="availableWith"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex="5"
        type="text"
        value=""
      />
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Issuing
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
        role="listbox"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Advising
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Issuing
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="field"
  >
    <label
      id="field_beneficiaryBankRole"
    >
      Beneficiary bank role
    </label>
    <div
      className="field"
    >
      <div
        className="ui radio checkbox"
        data-test-id="beneficiaryBankRole_ADVISING"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          onClick={[Function]}
          readOnly={true}
          tabIndex="6"
          type="radio"
        />
        <label>
          Advising bank
        </label>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui radio checkbox"
        data-test-id="beneficiaryBankRole_ISSUING"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          onClick={[Function]}
          readOnly={true}
          tabIndex="6"
          type="radio"
        />
        <label>
          Issuing bank
        </label>
      </div>
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Opening amount
    </label>
    <div
      className="ui right labeled input"
    >
      <input
        id="field_amount"
        name="amount"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex="7"
        type="text"
        value="1,000.00"
      />
      <div
        aria-expanded={false}
        className="ui dropdown label"
        name="currency"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        style={
          Object {
            "border": "1px solid rgb(226, 226, 226)",
          }
        }
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          USD
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              AED
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              CHF
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              EUR
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              GBP
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              JPY
            </span>
          </div>
          <div
            aria-checked={true}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              USD
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c0"
    data-test-id="amount-error"
  >
    
  </div>
  <div
    className="field"
    style={
      Object {
        "alignItems": "center",
        "display": "inherit",
        "justifyContent": "unset",
      }
    }
  >
    <label
      className="inputLabel"
    >
      Expiry date
    </label>
    <div
      className="ui input"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <input
        id="field_expiryDate"
        min="2017-02-14"
        name="expiryDate"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex="8"
        type="date"
        value="2019-10-11"
      />
    </div>
  </div>
  <div
    className="c0"
    data-test-id="expiryDate-error"
  >
    
  </div>
  <div
    className="field"
  >
    <label
      id="field_feesPayableBy"
    >
      Fees payable by
    </label>
    <div
      className="field"
    >
      <div
        className="ui radio checkbox"
        data-test-id="feesPayableBy_APPLICANT"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          onClick={[Function]}
          readOnly={true}
          tabIndex="9"
          type="radio"
        />
        <label>
          Applicant
        </label>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui radio checkbox"
        data-test-id="feesPayableBy_BENEFICIARY"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          onClick={[Function]}
          readOnly={true}
          tabIndex="9"
          type="radio"
        />
        <label>
          Beneficiary
        </label>
      </div>
    </div>
  </div>
  <div
    className="c0"
    data-test-id="feesPayableBy-error"
  >
    
  </div>
  <div
    className="field"
  >
    <label
      id="field_duplicateClause"
    >
      Duplicate clause
    </label>
    <div
      className="field"
    >
      <div
        className="ui checked radio checkbox"
        data-test-id="duplicateClause_YES"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={true}
          className="hidden"
          onClick={[Function]}
          readOnly={true}
          tabIndex="10"
          type="radio"
        />
        <label>
          Yes
        </label>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui radio checkbox"
        data-test-id="duplicateClause_NO"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          onClick={[Function]}
          readOnly={true}
          tabIndex="10"
          type="radio"
        />
        <label>
          No
        </label>
      </div>
    </div>
  </div>
  <div
    className="c0"
    data-test-id="duplicateClause-error"
  >
    
  </div>
  <div
    className="inline field"
    style={
      Object {
        "alignItems": "center",
        "display": "inherit",
        "justifyContent": "unset",
      }
    }
  >
    <label
      className="inputLabel"
    >
      Override standard template
    </label>
    <textarea
      className=""
      id="field_overrideStandardTemplate"
      name="overrideStandardTemplate"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onInput={[Function]}
      rows={3}
      style={
        Object {
          "height": "300px",
          "resize": "",
          "width": "100%",
        }
      }
      tabIndex="11"
      value="Default legal template clauses"
    />
  </div>
  <div
    className="c0"
    data-test-id="overrideStandardTemplate-error"
  >
    
  </div>
  <div
    className="inline field"
    style={
      Object {
        "alignItems": "center",
        "display": "inherit",
        "justifyContent": "unset",
      }
    }
  >
    <label
      className="inputLabel"
    >
      Additional information
    </label>
    <textarea
      className=""
      name="additionalInformation"
      onBlur={[Function]}
      onChange={[Function]}
      onInput={[Function]}
      rows={3}
      style={
        Object {
          "resize": "",
          "width": "100%",
        }
      }
      tabIndex="12"
      value="an additional info"
    />
  </div>
  <p>
    Additional information will be shared with all participants of the SBLC. It will not be included within the SBLC itself
  </p>
</form>
`;
